tests/integration/test_charm.py

```diff
@@ -166,6 +169,23 @@ async def test_settings_are_correct(ops_test: OpsTest, unit_id: int):
     assert unit.data["port-ranges"][0]["protocol"] == "tcp"
 
 
+@pytest.mark.group(1)
+async def test_postgresql_locales(ops_test: OpsTest) -> None:
+    raw_locales = await run_command_on_unit(
+        ops_test,
+        ops_test.model.applications[DATABASE_APP_NAME].units[0].name,
+        "ls /snap/charmed-postgresql/current/usr/lib/locale",
+    )
+    locales = raw_locales.splitlines()
+    locales.append("C")
+    locales.sort()
+
+    # Juju 2 has an extra empty element
+    if "" in locales:
+        locales.remove("")
+    assert locales == SNAP_LOCALES
+
+
```

Comment:
This should fail if the snap locales change after an update, so we can regenerate the SNAP_LOCALES

Reply:
If the `ls` command was failing/unstable before (when running inside the leader elected hook), wouldn't this test potentially be unstable for the same underlying reason? Shouldn't we maybe retry this command in case of failure? No hard opinion here, just worrying for instabilities :sweat_smile:

Reply:
> If the `ls` command was failing/unstable before (when running inside the leader elected hook), wouldn't this test potentially be unstable for the same underlying reason? Shouldn't we maybe retry this command in case of failure? No hard opinion here, just worrying for instabilities 😅

The original issue happens if we try to validate the config model while the snap is not yet available/failed to install, so we can't list the locales available within the snap. If we are reaching this far into the test suite, the snap has already been installed, since the cluster has already bootstrapped.

Reply:
Now I get the timing! Sounds good then